user_interface/app.rb

```diff
@@ -0,0 +1,58 @@
+require_relative '../components/book'
+require_relative '../components/rental'
+require_relative '../components/student'
+require_relative '../components/teacher'
+require_relative '../components/person'
+require_relative '../user_interface/books_ui'
+require_relative '../user_interface/person_ui'
+require_relative '../user_interface/rental_ui'
+
+class App
+  attr_accessor :books, :people, :rentals
+
+  def initialize
+    @books = BookUserInterface.new
+    @people = PersonUserInterface.new
+    @rentals = RentalUserInterface.new(@books, @people)
+  end
+
```

Comment:
- [ ] Please make sure that the data is preserved correctly. Currently, the data is preserved and saved while the app is opened. When the app is closed and reopened, you lose the previous session's data. As per [this](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_preserve_data.md#:~:text=On%20startup%20of%20your%20app%20load%20all%20data%20from%20files.) requirement.

Reply:
@ShadyShawkat please note that the required changes have been implemented as per the requirement